src/main/java/com/cityelf/controller/WaterForecastController.java

```diff
@@ -0,0 +1,77 @@
+package com.cityelf.controller;
+
+import com.cityelf.exceptions.ForecastAlreadyExistsException;
+import com.cityelf.exceptions.ForecastNotFoundException;
+import com.cityelf.model.Address;
+import com.cityelf.model.WaterForecast;
+import com.cityelf.service.WaterForecastService;
+import com.mysql.cj.core.exceptions.WrongArgumentException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+
+@RestController
+@RequestMapping("/waterForecast")
+public class WaterForecastController {
+
+  @Autowired
+  private WaterForecastService waterForecastService;
+
+  @RequestMapping("/all")
+  public Iterable<WaterForecast> getAll() {
+    return waterForecastService.getAll();
+  }
+
+  @RequestMapping(value = "/{wfId}", method = RequestMethod.GET)
+  public WaterForecast getWaterForecast(@PathVariable("wfId") long id) throws ForecastNotFoundException {
+    return waterForecastService.getForecast(id);
+  }
+
+  @RequestMapping(value = "/get", method = RequestMethod.GET)
+  public WaterForecast getWaterForecast(@RequestParam(name = "start") String startTime,
+      @RequestParam(name = "address") String address) throws ForecastNotFoundException {
+    return waterForecastService.getForecast(LocalDateTime.parse(startTime), address);
+  }
+
+  @RequestMapping(value = "/getByStart", method = RequestMethod.GET)
```

Comment:
The same as above for uppercase letters, let's get rid of them. Just **getbystart**